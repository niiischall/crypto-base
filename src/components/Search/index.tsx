import React, { useEffect, useState } from 'react';
import { Button } from '@material-ui/core';
import { useSelector } from 'react-redux';
import { makeStyles, withStyles } from '@material-ui/core/styles';
import { ArrowRightAlt } from '@material-ui/icons';

export interface Props {}

const useStyles = makeStyles((theme) => ({
  searchResultContainer: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    height: '100%',
  },
  searchHeader: {
    height: '10%',
    display: 'flex',
    alignItems: 'center',
    width: '100%',
  },
  searchHeading: {
    color: '#979797',
    display: 'flex',
    fontSize: 15,
    fontWeight: 700,
  },
  imageBox: {
    flex: 1,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  searchBox: {
    height: 200,
    borderRadius: '5px',
    backgroundColor: '#f8f8f8',
    boxShadow: '0 0 0 1px rgb(0 0 0 / 5%), 0 5px 15px rgb(0 0 0 / 15%)',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%',
    margin: '0px auto',
    padding: 15,
  },
  image: {
    width: 50,
    height: 50,
  },
  coin: {
    width: '75%',
    height: '100%',
  },
  name: {
    fontSize: 18,
    color: '#669900',
  },
  year: {
    display: 'block',
    margin: '5px 0px',
    fontSize: 10,
    color: '#f59842',
  },
  symbol: {
    fontSize: 10,
    fontWeight: 'bold',
    color: '#000',
    marginLeft: 5,
  },
  description: {
    display: 'block',
    width: '100%',
    fontSize: 12,
    color: '#000',
    lineHeight: '14.5px',
    margin: '5px 0px',
  },
}));

const StyledButton = withStyles((theme) => ({
  root: {
    textTransform: 'capitalize',
    textDecoration: 'underline',
    fontSize: 12,
    color: '#669900',
    padding: 0,
    '&:focus': {
      fontWeight: 700,
    },
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
  endIcon: {
    margin: 0,
  },
}))((props: any) => <Button {...props} />);

export const Search: React.FC<Props> = (props) => {
  const classes = useStyles();
  const searchedCoin = useSelector((state: any) => state.search.searchedCoin);

  const [coinDetails, setCoinDetails] = useState<any>({
    name: '',
    logo: '',
    symbol: '',
    urls: '',
    month: '',
    day: '',
    year: '',
    desc: '',
  });

  useEffect(() => {
    if (searchedCoin) {
      const coin: any = Object.values(searchedCoin)[0];
      const { name, logo, symbol, urls, date_added, description } = coin;
      const date = new Date(date_added);
      const [month, day, year] = [
        date.getMonth(),
        date.getDate(),
        date.getFullYear(),
      ];
      const desc = description.substr(0, 200);
      setCoinDetails({
        name,
        logo,
        symbol,
        urls,
        month,
        day,
        year,
        desc,
      });
    }
  }, [searchedCoin]);

  const navigateToKnowMore = (url: string) => {
    chrome.tabs.create({ url, active: false });
  };

  return (
    <div className={classes.searchResultContainer}>
      <div className={classes.searchHeader}>
        <span className={classes.searchHeading}>Search result</span>
      </div>
      <article className={classes.searchBox}>
        <div className={classes.imageBox}>
          <img
            src={coinDetails.logo}
            alt="coin-preview"
            className={classes.image}
          />
        </div>
        <div className={classes.coin}>
          <h3 className={classes.name}>
            {coinDetails.name}
            <span className={classes.symbol}>({coinDetails.symbol})</span>
          </h3>
          <span className={classes.year}>
            In circulation since{' '}
            <span style={{ fontWeight: 'bold' }}>
              {coinDetails.day}/{coinDetails.month}/{coinDetails.year}
            </span>
            .
          </span>
          <p className={classes.description}>{coinDetails.desc}...</p>
          <StyledButton
            onClick={() => navigateToKnowMore(coinDetails.urls.website[0])}
            endIcon={<ArrowRightAlt />}
          >
            Know More
          </StyledButton>
        </div>
      </article>
    </div>
  );
};

export default Search;
