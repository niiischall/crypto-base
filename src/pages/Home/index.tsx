import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  homeContainer: {
    flex: 1,
  },
}));

export const Home: React.FC<{}> = (props) => {
  const classes = useStyles();

  return (
    <div className={classes.homeContainer}>
      <span>&nbsp;</span>
    </div>
  );
};

export default Home;
