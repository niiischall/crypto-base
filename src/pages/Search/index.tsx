import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';

import { getTrendingCoinsDetails } from '../../store/actions/actionSearch';

const useStyles = makeStyles((theme) => ({
  searchContainer: {
    flex: 1,
    height: '100%',
    backgroundColor: '#eeeeee',
    overflowY: 'auto',
    paddingBottom: 15,
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flexWrap: 'wrap',
    paddingLeft: 15,
    paddingRight: 15,
    width: '100%',
  },
  searchHeader: {
    height: '10%',
    display: 'flex',
    alignItems: 'center',
    width: '100%',
  },
  searchHeading: {
    color: '#979797',
    display: 'flex',
    fontSize: 15,
    fontWeight: 700,
  },
  trendingCoin: {
    backgroundColor: 'white',
    width: '47.5%',
    height: 200,
    marginBottom: 15,
    padding: 10,
    boxShadow: '0px 2px 9px 0px rgb(0 0 0 / 2%)',
    borderRadius: '3px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  coin: {
    width: 50,
    height: 50,
    border: '2px solid #979797',
    borderRadius: '50%',
  },
  textBox: {
    margin: '5px 0px',
    textAlign: 'center',
  },
  name: {
    fontSize: 14,
    fontWeight: 600,
    marginBottom: 5,
  },
}));

export const Search: React.FC<{}> = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const trendingCoins = useSelector((state: any) => state.search.trendingCoins);
  const coinsDetails = useSelector((state: any) => state.search.coinsDetails);

  useEffect(() => {
    if (trendingCoins.length > 0 && !coinsDetails) {
      dispatch(getTrendingCoinsDetails(trendingCoins));
    }
  }, []);

  return (
    <div className={classes.searchContainer}>
      {trendingCoins.length > 0 &&
        coinsDetails &&
        Object.keys(trendingCoins).length > 0 && (
          <>
            <div className={classes.searchHeader}>
              <span className={classes.searchHeading}>Trending Now</span>
            </div>
            {trendingCoins.map((coin: any) => {
              const logo = coinsDetails[coin.id].logo;
              return (
                <div key={coin.id} className={classes.trendingCoin}>
                  <img
                    className={classes.coin}
                    src={logo}
                    alt={`${coin.slug}-logo`}
                  />
                  <div className={classes.textBox}>
                    <span className={classes.name}>{coin.name}</span>
                  </div>
                </div>
              );
            })}
          </>
        )}
    </div>
  );
};

export default Search;
