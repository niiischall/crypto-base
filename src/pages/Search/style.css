.search-container {
  flex: 1;
}