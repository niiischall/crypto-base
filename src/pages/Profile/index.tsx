import React from 'react';
import './style.css';

export const Profile: React.FC<{}> = (props) => {
  return (
    <div className="profile-container">
      <span>&nbsp;</span>
    </div>
  );
};

export default Profile;
