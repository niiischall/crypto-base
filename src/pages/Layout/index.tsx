import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Home from '../Home';
import Profile from '../Profile';
import Search from '../Search';

import { getFollowingCoinsDetails } from '../../store/actions/actionHome';

export interface Props {
  currentPage: string;
}

export const Layout: React.FC<Props> = (props) => {
  const { currentPage } = props;

  const dispatch = useDispatch();
  const followingCoins = useSelector((state: any) => state.home.followingCoins);

  useEffect(() => {
    if (followingCoins.length > 0) {
      dispatch(getFollowingCoinsDetails(followingCoins));
    }
  }, [dispatch, followingCoins]);

  return (
    <React.Fragment>
      {currentPage === 'home' && <Home />}
      {currentPage === 'search' && <Search />}
      {currentPage === 'profile' && <Profile />}
    </React.Fragment>
  );
};

export default Layout;
