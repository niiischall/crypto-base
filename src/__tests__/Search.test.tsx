import ReactDOM from 'react-dom';

import { Search } from '../pages/Search';

const props = {};

describe('Navigation Component', () => {
  const div = document.createElement('div');

  it('should match snapshot', () => {
    const component = ReactDOM.render(<Search {...props} />, div);
    expect(component).toMatchSnapshot();
  });
});
