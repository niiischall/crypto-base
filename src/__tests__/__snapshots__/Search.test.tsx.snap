// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should match snapshot 1`] = `null`;
